health-services/ExerciseSampleCompose/app/src/main/java/com/example/exercisesamplecompose/data/ExerciseClientManager.kt

```diff
@@ -96,20 +105,33 @@ class ExerciseClientManager @Inject constructor(
             )
         }
 
-        if (supportsDistanceMilestone(capabilities)) {
-            // Create a milestone goal. To make a milestone for every kilometer, set the initial
-            // threshold to 1km and the period to 1km.
```

Comment:
You could add something to your Thresholds data class, like isSet, so it becomes ... && thresholds.isSet which then reads a bit more logically than knowing that 0.0 means one has been entered.

You could also then create a default for your thresholds, and initialize to Thresholds.NOT_SET?

data class Thresholds {
  ...

  companion object {
    const val NOT_SET = Thresholds(0.0, 0.0)
  }
}